### PR TITLE
feat(gpu): implement PBSs that use new Hopper's features

### PR DESCRIPTION
**Pull request by pdroalves**
https://github.com/zama-ai/tfhe-rs-internal/issues/450
https://github.com/zama-ai/concrete-internal/issues/345

closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Review by agnesLeroy — COMMENTED** [review]
Thanks a lot for this @pdroalves, I've launched the integer and core crypto benchmarks to see if there is an effect on performance but I realize now this only affects H100, so the benches have to be launched manually...

**Comment by agnesLeroy** [comment]
@pdroalves when the integer multi-bit benches have run, could you paste the results here please? Would be good to have pbs throughput benchmarks as well, PR on the way to add them in the code base -> https://github.com/zama-ai/tfhe-rs/pull/936

**Comment by agnesLeroy** [comment]
@pdroalves it'll be better to run the benchmarks on hyperstack I think, much cheaper. Also to really support this we should have the CI run on H100 regularly... it'll be soon I hope.

Another thing I'm thinking of, is that we could try to do the same for the classical bootstrap, and see if it improves the throughput. It'd be really interesting to see the effect there.

**Comment by agnesLeroy** [comment]
@pdroalves you can rebase on main to get the core crypto throughput benchmarks (there are 8192 inputs in the bench, total time is measured by criterion).

File: backends/tfhe-cuda-backend/cuda/src/pbs/bootstrap_tbc_multibit.cuh [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        lwe_array_in, lwe_input_indexes, keybundle_fft, buffer_fft,
+        global_accumulator, lwe_dimension, glwe_dimension, polynomial_size,
+        base_log, level_count, grouping_factor, lwe_offset, chunk_size,
+        keybundle_size_per_input));
+  }
+}
+
+template <typename Torus, class params>
+__host__ bool
+verify_cuda_bootstrap_tbc_multi_bit_grid_size(int glwe_dimension,
```
[the remark this answers is not in the document]
Comment (pdroalves) [reply backends/tfhe-cuda-backend/cuda/src/pbs/bootstrap_tbc_multibit.cuh:293]: Yeah, I was mimicking what we have with the "fast" variant, but this time we need to check that during compiling time so the approach should be other.

**Pull request closed without merging**